### PR TITLE
R fix dependency installation

### LINKED ISSUE #3473: R package linting does not install dependencies raising errors
## Describe the bug

For the R language linter, when linting a package, the package is getting installed but it's dependencies do not, hence raising lint errors

## Expected behavior

Would expect that the package itself gets installed and also the dependencies

## Steps to Reproduce

I have created a public repo to reproduce this: https://github.com/tonyk7440/superlintertest

Can see log here:
https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538

Relevant parts
```
2022-10-21 15:11:02 [INFO]   Detected R Language files to lint.
2022-10-21 15:11:02 [INFO]   Trying to install the R package inside:[/github/workspace]
2022-10-21 15:11:03 [WARN]   ERROR: Failed to install the build package at:[superlintertest_0.1.0.tar.gz]
2022-10-21 15:11:03 [WARN]   INSTALL_CMD:[* installing to library ‘/usr/lib/R/library’
ERROR: dependency ‘dplyr’ is not available for package ‘superlintertest’
* removing ‘/usr/lib/R/library/superlintertest’]
2022-10-21 15:11:03 [INFO]   
```

and 
```
2022-10-21 15:11:15 [INFO]   File:[/github/workspace/R/pull_column.R]
2022-10-21 15:11:[16](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:17) [INFO]    - File:[pull_column.R] was linted with [R] successfully
2022-10-21 15:11:16 [INFO]      - Command output:
------
Warning: file=/github/workspace/R/pull_column.R,line=14,col=5,[namespace_linter] Package 'dplyr' is not installed.
------
```

## Logs

```
Run github/super-linter/slim@v4
/usr/bin/docker run --name ghcriogithubsuperlinterslimv497_5aacfa --label 8d5581 --workdir /github/workspace --rm -e "R_LIBS_USER" -e "TZ" -e "_R_CHECK_SYSTEM_CLOCK_" -e "NOT_CRAN" -e "RSPM" -e "VALIDATE_ALL_CODEBASE" -e "DEFAULT_BRANCH" -e "GITHUB_TOKEN" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/superlintertest/superlintertest":"/github/workspace" ghcr.io/github/super-linter:slim-v4.9.7
--------------------------------------------------------------------------------

                              /@@#///////@@/(@//@%/(@.@(       @@
                          @@//////////////////////////////#*  @@@
                        @////@//(///////////@@@@@///@//@/@**//@@(
                      @///////@///////////////@@@@    (           @,
                     @/(&/@////////////////////                     @
                    @////////////////////////@@                      @
                  @%////////(//////////%/////&@            @@       *,@           ______________
             @@@@@/@/#/////(&//////////////////                       .@         /              \
        *@@@@@.    .%///(//@//////////////////&.   .@@,                 @%      / Don't mind me  \
      @@%           .&@&&/@.@//&/////(//////////    @@@@@@@@@         .. &@    / I'm just looking \
    @@%               @@@@@   @&/////////////////#   @/       V  @@/ ,@@@ @   <  for some trash... |
@@@%                   @@@@        .%@@@@//////#@ @   @@         @     .,.     \__________________/
                                          @@@/@(  (@@@@% @/\      %
                                           @@@@(    .     .@@/\   #
                                             @                  %@%

--------------------------------------------------------------------------------
2022-10-21 15:10:57 [INFO]   ---------------------------------------------
2022-10-21 15:10:57 [INFO]   --- GitHub Actions Multi Language Linter ----
2022-10-21 15:10:57 [INFO]    - Image Creation Date:[2022-10-04T14:29:23Z]
2022-10-21 15:10:57 [INFO]    - Image Revision:[b5715650a7bd14dad5e74e71b7726e50d929903b]
2022-10-21 15:10:57 [INFO]    - Image Version:[b5715650a7bd14dad5e74e71b7726e50d929903b]
2022-10-21 15:10:57 [INFO]   ---------------------------------------------
2022-10-21 15:10:57 [INFO]   ---------------------------------------------
2022-10-21 15:10:57 [INFO]   The Super-Linter source code can be found at:
2022-10-21 15:10:57 [INFO]    - https://github.com/github/super-linter
2022-10-21 15:10:57 [INFO]   ---------------------------------------------
Removing Languages from LANGUAGE_ARRAY for slim image...
found item:[ARM], removing Language...
found item:[CSHARP], removing Language...
found item:[ENV], removing Language...
found item:[POWERSHELL], removing Language...
found item:[RUST_2015], removing Language...
found item:[RUST_2018], removing Language...
found item:[RUST_2021], removing Language...
found item:[RUST_CLIPPY], removing Language...
Removing Linters from LINTER_NAMES_ARRAY for slim image...
found item:[ARM], removing linter...
found item:[CSHARP], removing linter...
found item:[ENV], removing linter...
found item:[POWERSHELL], removing linter...
found item:[RUST_2015], removing linter...
found item:[RUST_2018], removing linter...
found item:[RUST_2021], removing linter...
found item:[RUST_CLIPPY], removing linter...
2022-10-21 15:10:57 [INFO]   --------------------------------------------
2022-10-21 15:10:57 [INFO]   Gathering GitHub information...
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_SHA], value:[9798[11](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:12)6e017f52042b27b1a1024d9914f37d7cea]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_WORKSPACE], value:[/github/workspace]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_EVENT_PATH], value:[/github/workflow/event.json]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_ORG], value:[tonyk7440]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_REPO], value:[superlintertest]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_TOKEN]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_REPOSITORY], value:[tonyk7440/superlintertest]
2022-10-21 15:10:57 [INFO]   Successfully found:[GITHUB_RUN_ID], value:[3298470514]
2022-10-21 15:10:57 [INFO]   --------------------------------------------
2022-10-21 15:10:57 [INFO]   Gathering user validation information...
2022-10-21 15:10:57 [INFO]   - Validating ALL files in code base...
2022-10-21 15:11:00 [INFO]   ----------------------------------------------
2022-10-21 15:11:00 [INFO]   User provided file:[/github/workspace/.github/linters/.flake8] exists, setting rules file...
2022-10-21 15:11:00 [INFO]   ----------------------------------------------
2022-10-21 15:11:00 [INFO]   User provided file:[/github/workspace/.github/linters/.lintr] exists, setting rules file...
2022-10-21 15:11:01 [INFO]   ---------------------------------
2022-10-21 15:11:01 [INFO]   ------ File list to check: ------
2022-10-21 15:11:01 [INFO]   ---------------------------------
2022-10-21 15:11:02 [INFO]   ----------------------------------------------
2022-10-21 15:11:02 [INFO]   Successfully gathered list of files...
2022-10-21 15:11:02 [INFO]   Detected R Language files to lint.
2022-10-21 15:11:02 [INFO]   Trying to install the R package inside:[/github/workspace]
2022-10-21 15:11:03 [WARN]   ERROR: Failed to install the build package at:[superlintertest_0.1.0.tar.gz]
2022-10-21 15:11:03 [WARN]   INSTALL_CMD:[* installing to library ‘/usr/lib/R/library’
ERROR: dependency ‘dplyr’ is not available for package ‘superlintertest’
* removing ‘/usr/lib/R/library/superlintertest’]
2022-10-21 15:11:03 [INFO]   
2022-10-21 15:11:03 [INFO]   ----------------------------------------------
2022-10-21 15:11:03 [INFO]   ----------------------------------------------
2022-10-21 15:11:03 [INFO]   Linting [GITHUB_ACTIONS] files...
2022-10-21 15:11:03 [INFO]   ----------------------------------------------
2022-10-21 15:11:03 [INFO]   ----------------------------------------------
2022-10-21 15:11:03 [INFO]   ---------------------------
2022-10-21 15:11:03 [INFO]   File:[/github/workspace/.github/workflows/super_linter.yml]
2022-10-21 15:11:03 [INFO]    - File:[super_linter.yml] was linted with [actionlint] successfully
2022-10-21 15:11:03 [INFO]   
2022-10-21 15:11:03 [INFO]   ----------------------------------------------
2022-10-21 15:11:03 [INFO]   ----------------------------------------------
2022-10-21 15:11:03 [INFO]   Linting [GITLEAKS] files...
2022-10-21 15:11:04 [INFO]   ----------------------------------------------
2022-10-21 15:11:04 [INFO]   ----------------------------------------------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/.github/.gitignore]
2022-10-21 15:11:04 [INFO]    - File:[.gitignore] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.85ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/.github/linters/.flake8]
2022-10-21 15:11:04 [INFO]    - File:[.flake8] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.98ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/.github/linters/.lintr]
2022-10-21 15:11:04 [INFO]    - File:[.lintr] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.41ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/.github/workflows/super_linter.yml]
2022-10-21 15:11:04 [INFO]    - File:[super_linter.yml] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.84ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/.gitignore]
2022-10-21 15:11:04 [INFO]    - File:[.gitignore] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.17ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/DESCRIPTION]
2022-10-21 15:11:04 [INFO]    - File:[DESCRIPTION] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.31ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/NAMESPACE]
2022-10-21 15:11:04 [INFO]    - File:[NAMESPACE] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.51ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/R/hello.R]
2022-10-21 15:11:04 [INFO]    - File:[hello.R] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.69ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/R/pull_column.R]
2022-10-21 15:11:04 [INFO]    - File:[pull_column.R] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.15ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/README.md]
2022-10-21 15:11:04 [INFO]    - File:[README.md] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 10.3ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/man/hello.Rd]
2022-10-21 15:11:04 [INFO]    - File:[hello.Rd] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.23ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/man/pull_column.Rd]
2022-10-21 15:11:04 [INFO]    - File:[pull_column.Rd] was linted with [gitleaks] successfully
2022-10-21 15:11:04 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.16ms
3:11PM INF no leaks found
------
2022-10-21 15:11:04 [INFO]   ---------------------------
2022-10-21 15:11:04 [INFO]   File:[/github/workspace/superlintertest.Rproj]
2022-10-21 15:11:05 [INFO]    - File:[superlintertest.Rproj] was linted with [gitleaks] successfully
2022-10-21 15:11:05 [INFO]      - Command output:
------

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

3:11PM INF scan completed in 9.57ms
3:11PM INF no leaks found
------
2022-10-21 15:11:05 [INFO]   
2022-10-21 15:11:05 [INFO]   ----------------------------------------------
2022-10-21 15:11:05 [INFO]   ----------------------------------------------
2022-10-21 15:11:05 [INFO]   Linting [JSCPD] files...
2022-10-21 15:11:05 [INFO]   ----------------------------------------------
2022-10-21 15:11:05 [INFO]   ----------------------------------------------
2022-10-21 15:11:05 [INFO]   ---------------------------
2022-10-21 15:11:05 [INFO]   File:[/github/workspace/.github/.gitignore]
2022-10-21 15:11:06 [INFO]    - File:[.gitignore] was linted with [jscpd] successfully
2022-10-21 15:11:06 [INFO]      - Command output:
------
Detection time:: 1.032ms
------
2022-10-21 15:11:06 [INFO]   ---------------------------
2022-10-21 15:11:06 [INFO]   File:[/github/workspace/.github/linters/.flake8]
2022-10-21 15:11:06 [INFO]    - File:[.flake8] was linted with [jscpd] successfully
2022-10-21 15:11:06 [INFO]      - Command output:
------
Detection time:: 1.147ms
------
2022-10-21 15:11:06 [INFO]   ---------------------------
2022-10-21 15:11:06 [INFO]   File:[/github/workspace/.github/linters/.lintr]
2022-10-21 15:11:07 [INFO]    - File:[.lintr] was linted with [jscpd] successfully
2022-10-21 15:11:07 [INFO]      - Command output:
------
Detection time:: 1.139ms
------
2022-10-21 15:11:07 [INFO]   ---------------------------
2022-10-21 15:11:07 [INFO]   File:[/github/workspace/.github/workflows/super_linter.yml]
2022-10-21 15:11:07 [INFO]    - File:[super_linter.yml] was linted with [jscpd] successfully
2022-10-21 15:11:07 [INFO]      - Command output:
------
Found 0 clones.
Detection time:: 13.87ms
------
2022-10-21 15:11:07 [INFO]   ---------------------------
2022-10-21 15:11:07 [INFO]   File:[/github/workspace/.gitignore]
2022-10-21 15:11:08 [INFO]    - File:[.gitignore] was linted with [jscpd] successfully
2022-10-21 15:11:08 [INFO]      - Command output:
------
Detection time:: 1.011ms
------
2022-10-21 15:11:08 [INFO]   ---------------------------
2022-10-21 15:11:08 [INFO]   File:[/github/workspace/DESCRIPTION]
2022-10-21 15:11:08 [INFO]    - File:[DESCRIPTION] was linted with [jscpd] successfully
2022-10-21 15:11:08 [INFO]      - Command output:
------
Detection time:: 1.036ms
------
2022-10-21 15:11:08 [INFO]   ---------------------------
2022-10-21 15:11:08 [INFO]   File:[/github/workspace/NAMESPACE]
2022-10-21 15:11:09 [INFO]    - File:[NAMESPACE] was linted with [jscpd] successfully
2022-10-21 15:11:09 [INFO]      - Command output:
------
Detection time:: 1.027ms
------
2022-10-21 15:11:09 [INFO]   ---------------------------
2022-10-21 15:11:09 [INFO]   File:[/github/workspace/R/hello.R]
2022-10-21 15:11:10 [INFO]    - File:[hello.R] was linted with [jscpd] successfully
2022-10-21 15:11:10 [INFO]      - Command output:
------
Found 0 clones.
Detection time:: 9.398ms
------
2022-10-21 15:11:10 [INFO]   ---------------------------
2022-10-21 15:11:10 [INFO]   File:[/github/workspace/R/pull_column.R]
2022-10-21 15:11:10 [INFO]    - File:[pull_column.R] was linted with [jscpd] successfully
2022-10-21 15:11:10 [INFO]      - Command output:
------
Found 0 clones.
Detection time:: 9.542ms
------
2022-10-21 15:11:10 [INFO]   ---------------------------
2022-10-21 15:11:10 [INFO]   File:[/github/workspace/README.md]
2022-10-21 15:11:11 [INFO]    - File:[README.md] was linted with [jscpd] successfully
2022-10-21 15:11:11 [INFO]      - Command output:
------
Found 0 clones.
Detection time:: 13.746ms
------
2022-10-21 15:11:11 [INFO]   ---------------------------
2022-10-21 15:11:11 [INFO]   File:[/github/workspace/man/hello.Rd]
2022-10-21 15:11:11 [INFO]    - File:[hello.Rd] was linted with [jscpd] successfully
2022-10-21 15:11:11 [INFO]      - Command output:
------
Detection time:: 1.063ms
------
2022-10-21 15:11:11 [INFO]   ---------------------------
2022-10-21 15:11:11 [INFO]   File:[/github/workspace/man/pull_column.Rd]
2022-10-21 15:11:[12](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:13) [INFO]    - File:[pull_column.Rd] was linted with [jscpd] successfully
2022-10-21 15:11:12 [INFO]      - Command output:
------
Detection time:: 1.023ms
------
2022-10-21 15:11:12 [INFO]   ---------------------------
2022-10-21 15:11:12 [INFO]   File:[/github/workspace/superlintertest.Rproj]
2022-10-21 15:11:12 [INFO]    - File:[superlintertest.Rproj] was linted with [jscpd] successfully
2022-10-21 15:11:12 [INFO]      - Command output:
------
Detection time:: 1.052ms
------
2022-10-21 15:11:[13](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:14) [INFO]   
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   Linting [MARKDOWN] files...
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   ---------------------------
2022-10-21 15:11:13 [INFO]   File:[/github/workspace/README.md]
2022-10-21 15:11:13 [ERROR]   Found errors in [markdownlint] linter!
2022-10-21 15:11:13 [ERROR]   Error code: 1. Command output:
------
/github/workspace/README.md:3 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Install"]
/github/workspace/README.md:5 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
------
2022-10-21 15:11:13 [INFO]   
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   Linting [NATURAL_LANGUAGE] files...
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   ----------------------------------------------
2022-10-21 15:11:13 [INFO]   ---------------------------
2022-10-21 15:11:13 [INFO]   File:[/github/workspace/README.md]
2022-10-21 15:11:[14](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:15) [INFO]    - File:[README.md] was linted with [textlint] successfully
2022-10-21 [15](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:16):11:14 [INFO]   No .lintr configuration file found, using defaults.
2022-10-21 15:11:14 [INFO]   
2022-10-21 15:11:14 [INFO]   ----------------------------------------------
2022-10-21 15:11:14 [INFO]   ----------------------------------------------
2022-10-21 15:11:14 [INFO]   Linting [R] files...
2022-10-21 15:11:14 [INFO]   ----------------------------------------------
2022-10-21 15:11:14 [INFO]   ----------------------------------------------
2022-10-21 15:11:14 [INFO]   ---------------------------
2022-10-21 15:11:14 [INFO]   File:[/github/workspace/R/hello.R]
2022-10-21 15:11:15 [INFO]    - File:[hello.R] was linted with [R] successfully
2022-10-21 15:11:15 [INFO]   ---------------------------
2022-10-21 15:11:15 [INFO]   File:[/github/workspace/R/pull_column.R]
2022-10-21 15:11:[16](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:17) [INFO]    - File:[pull_column.R] was linted with [R] successfully
2022-10-21 15:11:16 [INFO]      - Command output:
------
Warning: file=/github/workspace/R/pull_column.R,line=14,col=5,[namespace_linter] Package 'dplyr' is not installed.
------
2022-10-21 15:11:[17](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:18) [INFO]   
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   Linting [YAML] files...
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   ---------------------------
2022-10-21 15:11:17 [INFO]   File:[/github/workspace/.github/workflows/super_linter.yml]
2022-10-21 15:11:17 [INFO]    - File:[super_linter.yml] was linted with [yamllint] successfully
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   The script has completed
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:17 [INFO]   ----------------------------------------------
2022-10-21 15:11:[18](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:19) [ERROR]   ERRORS FOUND in MARKDOWN:[1]
[20](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:21)22-10-[21](https://github.com/tonyk7440/superlintertest/actions/runs/3298470514/jobs/5440630538#step:6:22) 15:11:20 [FATAL]   Exiting with errors found!
```

### PR DESCRIPTION
**Pull request by tonyk7440**
Fixes #3473 
I tried using the updated action [here](https://github.com/tonyk7440/superlintertest/actions/runs/3347436506/jobs/5545651520) to test the updates but it doesn't seem to be retrieving them. I believe it should fix the problem but I am unsure how to test that to make sure. Any advice here is appreciated

## Proposed Changes

1. Installs the remotes package
2. Installs the current package and its dependencies using the remotes package

### Author/Contributor
- [X] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by tonyk7440** [comment]
Not stale, waiting on review of PR

File: lib/functions/detectFiles.sh [hunk 1]
```diff
@@ -479,7 +479,8 @@ function RunAdditionalInstalls() {
       ##############################
       INSTALL_CMD=$(
         cd "${WORKSPACE_PATH}" || exit 0
-        R CMD INSTALL "${BUILD_PKG}" 2>&1
+        R -e "install.packages('remotes', repos = 'https://cloud.r-project.org/')"
```
Comment (ferrarimarco) [inline lib/functions/detectFiles.sh:482]: Can you redirect stderr to stdout here as well? So we get the complete output when debugging

Comment (tonyk7440) [reply lib/functions/detectFiles.sh]: @ferrarimarco Done now I think

File: lib/functions/detectFiles.sh [hunk 2]
```diff
@@ -479,7 +479,8 @@ function RunAdditionalInstalls() {
       ##############################
       INSTALL_CMD=$(
         cd "${WORKSPACE_PATH}" || exit 0
-        R CMD INSTALL "${BUILD_PKG}" 2>&1
+        R -e "install.packages('remotes', repos = 'https://cloud.r-project.org/')" 2>&1
+        R -e "remotes::install_local('.', dependencies=T)" 2>&1
       )
```
Comment (ferrarimarco) [inline lib/functions/detectFiles.sh:485]: Thanks for the update.

I think the build is complaining.

From the [log](https://github.com/github/super-linter/actions/runs/3875288592/jobs/6608799248#step:9:2686):

```
2023-01-09 16:47:44 [INFO]   Trying to install the R package inside:[/tmp/lint/.automation/test]
2023-01-09 16:47:45 [WARN]   ERROR! Failed to run:[R CMD build] at location:[/tmp/lint/.automation/test]
2023-01-09 16:47:45 [WARN]   BUILD_CMD:[* checking for file ‘/tmp/lint/.automation/test/DESCRIPTION’ ... NO]
```

We should make the build fail in that case so we get noticed of this issue during tests.

(Apparently, I can't request changes on lines you didn't modify from the GitHub GUI, so I'll write the details here directly.)

Can you please change the following check, starting on line 494 from:

```shell
      if [ "${ERROR_CODE}" -ne 0 ]; then
        # Error
        warn "ERROR: Failed to install the build package at:[${BUILD_PKG}]"
        warn "INSTALL_CMD:[${INSTALL_CMD}]"
      fi
```

to:

```shell
      debug "INSTALL_CMD:[${INSTALL_CMD}]"
      if [ "${ERROR_CODE}" -ne 0 ]; then
        fatal "ERROR: Failed to install the build package at:[${BUILD_PKG}]"
      fi
```

After that, we can troubleshoot the failure.

Thanks!

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

File: lib/functions/detectFiles.sh (hunk 2 above)
Comment (tonyk7440) [reply lib/functions/detectFiles.sh:485]: Hi @ferrarimarco , thanks, just updated the code according to your comment

**Comment by tonyk7440** [comment]
Hi @ferrarimarco , could you perhaps review again?

Thanks!

**Review by ferrarimarco — APPROVED** [review]

**Pull request merged**